frontend/src/lib/derived/main-transaction-fee.derived.ts

```
@@ -1,8 +1,16 @@
-import { transactionsFeesStore } from "$lib/stores/transaction-fees.store";
-import { ICPToken, TokenAmount } from "@dfinity/utils";
+import { OWN_CANISTER_ID_TEXT } from "$lib/constants/canister-ids.constants";
+import { NNS_TOKEN_DATA } from "$lib/constants/tokens.constants";
+import { tokensStore } from "$lib/stores/tokens.store";
+import { TokenAmount, isNullish } from "@dfinity/utils";
 import { derived } from "svelte/store";
 
-export const mainTransactionFeeStoreAsToken = derived(
-  transactionsFeesStore,
-  ($store) => TokenAmount.fromE8s({ amount: $store.main, token: ICPToken })
-);
+export const mainTransactionFeeStoreAsToken = derived(tokensStore, ($store) => {
+  const icpToken = $store[OWN_CANISTER_ID_TEXT].token;
+  if (isNullish(icpToken)) {
```

Is this necessary?

---

For TS reasons yes. But not really because the `$store` will have the hardcoded value.

I thought it was better to return the hard-coded value than raise an error here though.

---

Error raised.